dspace-api/src/test/java/org/dspace/eperson/EPersonInWorkflowTest.java

```diff
@@ -0,0 +1,1477 @@
+                                                                                       .getXmlWorkflowItemService();
+    protected WorkflowService workflowService = WorkflowServiceFactory.getInstance().getWorkflowService();
+    protected WorkspaceItemService workspaceItemService = ContentServiceFactory.getInstance()
+                                                                               .getWorkspaceItemService();
+    protected ClaimedTaskService claimedTaskService = XmlWorkflowServiceFactory.getInstance().getClaimedTaskService();
+    protected PoolTaskService poolTaskService = XmlWorkflowServiceFactory.getInstance().getPoolTaskService();
+
+    protected XmlWorkflowService xmlWorkflowService = XmlWorkflowServiceFactory.getInstance().getXmlWorkflowService();
+    protected CollectionRoleService collectionRoleService = XmlWorkflowServiceFactory.getInstance()
+                                                                                     .getCollectionRoleService();
+
+
+    private EPerson workflowUserA;
+    private EPerson workflowUserB;
+    private EPerson workflowUserC;
+    private EPerson workflowUserD;
+
+    private static final Logger log = org.apache.logging.log4j.LogManager.getLogger(EPersonInWorkflowTest.class);
+
+    /**
+     * This method will be run before every test as per @Before. It will
+     * initialize resources required for the tests.
+     *
+     * Other methods can be annotated with @Before here or in subclasses but no
+     * execution order is guaranteed
+     */
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        context.turnOffAuthorisationSystem();
+
+        workflowUserA = EPersonBuilder.createEPerson(context).withEmail("workflowUserA@example.org").build();
+        workflowUserB = EPersonBuilder.createEPerson(context).withEmail("workflowUserB@example.org").build();
+        workflowUserC = EPersonBuilder.createEPerson(context).withEmail("workflowUserC@example.org").build();
+        workflowUserD = EPersonBuilder.createEPerson(context).withEmail("workflowUserD@example.org").build();
+
+        context.restoreAuthSystemState();
+
+    }
+
+
+    /**
+     * This test has the following setup:
+     * - Step 1: user B
+     * - Step 2: user C
+     * - Step 3: user B
+     *
+     * This test will perform the following checks:
+     * - create a workspace item, and let it move to step 1
+     * - approve it by user B
+     * - verify that the item moved to step 2
+     * - Claim it by user C
+     * - delete user B
+     * - verify the delete is refused
+     * - remove user B from step 3
+     * - approve it by user C
+     * - verify that the item is archived without any actions apart from removing user B
+     * - delete user B
+     * - verify the delete is refused
+     * - remove user B from step 1
+     * - delete user B
+     * - verify the delete succeeds
```

Comment:
@benbosman : The scenario above doesn't make total sense to me. I don't see why deletion of user B must fail until they are removed from both approval groups?   Why wouldn't a deletion just remove them from those approval groups (and let the approval proceed with other users in those groups)?

Does this have to do with whether a EPerson is the *only* member of a workflow role Group?

Simply put, I think we need to think about this more:
1. If an EPerson is only one (of many) member of a workflow role Group, then it seems like deleting them should just remove them from the group.  I.e. no error should occur.
2. If an EPerson is the *only* member of a workflow role Group, then we have two options:  
    1. We delete both the EPerson & the Workflow Role Group (i.e. you are deleting that workflow step by deleting the only member of it).  We could warn an Admin here if needed by saying: "By deleting this person, you will also remove these workflow steps of which they are the only member: "
    2. We refuse to delete the EPerson until they are removed from each of the Workflow Role Groups.  This is easier for us, but potentially more work for the DSpace Admin, who may have to go in and edit/delete the EPerson from a number of Groups one by one.

Reply:
I now realize (after you added more details on the new "rules" here to the PR description) that my above comment is a misunderstanding of this test code.  I had previously (wrongly) assumed this test was verifying that "no workflow group can be empty".

But, now I see that this test is likely verifying that the user cannot be removed *if* they are the only member of a Workflow Group that has items assigned.

I think this PR would be **much easier** to review if these tests all documented the scenario they are trying to tests.  So, while the list of checks is useful here, it'd be nice to summarize them at the top by saying something like `"This test verifies that an EPerson cannot be removed if they are the only member of a Workflow Group that has tasks currently assigned to it."`

So, please disregard my comments above..the only action here is to add a summary comment at the top of each test to summarize the scenario you are testing for.

Reply:
Most of this prior feedback can be ignored, *except* as noted, it'd be useful to update the comments above these tests to simply summarize the scenario that each test is attempting to verify.  The detailed list of steps is sometimes confusing to follow.  So, I *think* this test could say something like: `This test verifies that an EPerson cannot be removed if they are the only member of a Workflow Group that has tasks currently assigned to it.`

The same is true for all tests in this class...it's sometimes hard to follow exactly why they are performing the steps they are performing. So, a simple summary sentence would help describe the scenario much more clearly.

Reply:
AFAIK this has already been done as well

Reply:
@benbosman : I'm not seeing the requested changes to this `EPersonInWorkflowTest` class methods.

Reply:
I'd still appreciate simple summary notes at the beginning of each of these tests.  It's very hard to follow these tests & the detailed "This test will perform the following checks:" does not always make it clear which scenario is being tested for.  

As I noted previously, I *think* this test method can be easily summarized as `"This test verifies that an EPerson cannot be removed if they are the only member of a Workflow Group that has tasks currently assigned to it."`  However, I'm not sure that's accurate, as it also seems like it might be testing that a user with claimed tasks cannot be deleted?

I think these tests would be a lot easier to understand if the JavaDocs just had a brief summary of the scenarios tested, and you moved the step-by-step summary into *inline comments*.  Otherwise, I'm stuck reading a line and then having to scroll up to the JavaDocs to see which step it might be referencing, then reading the next line, and scroll back up, etc.

It's also extremely difficult determining the differences between  `testDeleteUserWhenOnlyUserInGroup1`, `testDeleteUserWhenOnlyUserInGroup2`, ``testDeleteUserWhenOnlyUserInGroup3`...all the way to 7.  Again, I think it'd be easier to understand if the JavaDocs had a simple summary of what makes this tests *unique*, and you move more of the details to inline (to describe what each line is doing). 

I realize this would take work, but without these changes, this test class is very hard to review.  I've taken 30mins to just check the first method looks correct (it does). But, I don't know how it differs from all the rest yet.  (And, at a glance many of these test methods look like they run the same check over and over again...for instance, each one keeps checking that User B cannot be deleted if they are the last member of the first step group... it seems like we only need to make that check *once*)